### PR TITLE
Remove SecurityManager Logger TimeTester singletones

### PR DESCRIPTION
**Pull request by Kozoriz**
Removed singletones from SecuriyManager, Logger and TimeTester classes.
TimeTester renamed to TelemetryMonitor.
Corrected Unit Tests after changes.

### CODE REVIEW

**Comment by Kozoriz** [comment]
Please review : @AByzhynar , @AMikhniev , @anosach-luxoft , @VVeremjova , @LuxoftAKutsan

**Comment by AMikhniev** [comment]
Reviewed

File: src/3rd_party-static/gmock-1.7.0/src/gmock-spec-builders.cc [hunk 1]
```diff
@@ -843,7 +847,7 @@ bool Mock::AsyncVerifyAndClearExpectationsLocked(int timeout_msec)
       // Get finial result and clear expectation
       const bool final_verification = base->VerifyAndClearExpectationsLocked();
       if (!final_verification) {
-        expectations_met = false;
+//        expectations_met = false;
```
Comment (LuxoftAKutsan) [inline src/3rd_party-static/gmock-1.7.0/src/gmock-spec-builders.cc]: @Kozoriz Why code commented but not deleted?

File: src/appMain/life_cycle.cc [hunk 2]
```diff
@@ -381,19 +381,12 @@ void LifeCycle::StopComponents() {
   connection_handler_->Stop();
 
   LOG4CXX_INFO(logger_, "Destroying Protocol Handler");
+  DCHECK_OR_RETURN_VOID(protocol_handler_);
   delete protocol_handler_;
 
   LOG4CXX_INFO(logger_, "Destroying Connection Handler.");
   delete connection_handler_;
 
-#ifdef ENABLE_SECURITY
```
Comment (LuxoftAKutsan) [inline src/appMain/life_cycle.cc]: Why this changes are not in security manager commit ?

File: src/components/utils/src/push_log.cc [hunk 3]
```diff
@@ -72,8 +75,25 @@ bool logs_enabled() {
...
+}
+
+void create_log_message_loop_thread() {
+  if (!log_message_loop_thread) {
+    log_message_loop_thread = new LogMessageLoopThread();
+  }
+}
+
+void delete_log_message_loop_thread() {
+    delete log_message_loop_thread;
```
Comment (LuxoftAKutsan) [inline src/components/utils/src/push_log.cc]: @Kozoriz Would be nice to  set log_message_loop_thread to NULL

**Comment by LuxoftAKutsan** [comment]
reviewed

**Comment by AByzhynar** [comment]
Reviewed

File: src/appMain/life_cycle.cc (hunk 2 above)
Comment (Kozoriz) [inline src/appMain/life_cycle.cc]: @LuxoftAKutsan , this section found when i check this commit. This section creates core dump.

File: src/3rd_party-static/gmock-1.7.0/src/gmock-spec-builders.cc (hunk 1 above)
Comment (Kozoriz) [inline src/3rd_party-static/gmock-1.7.0/src/gmock-spec-builders.cc]: @LuxoftAKutsan , corrected in bf1ca4c

File: src/components/utils/src/push_log.cc (hunk 3 above)
Comment (Kozoriz) [inline src/components/utils/src/push_log.cc]: @LuxoftAKutsan , done 52703cf
